### ISSUE TITLE
[BUG] Exception thrown while reading page; System.Net.WebException

### ISSUE THREAD

**Issue by JourneyOver** [Depressurizer/Depressurizer#129]
**Describe the bug**

Scraping game info eventually hits a point where it just seems to halt completely no matter how long you let it sit there trying to scrap game information.

This is the full error code in the log.

```
Exception thrown while reading page; System.Net.WebException: The operation has timed out
   at System.Net.HttpWebRequest.GetResponse()
   at Depressurizer.Core.Models.DatabaseEntry.ScrapeStoreHelper(String languageCode).
```

**To Reproduce**
Steps to reproduce the behavior (With a brand new `database.json` file) :
1. Go to the database editor
2. Tick off `show owned only`
3. Click `Scrape Unscraped`
4. Wait a while
5. ???
6. See Error in log file.

**Expected behavior**

For it to continue scraping all the unscraped games.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10 x64 bit Pro, Update 1809 (all the way up to date)
 - Version: 4.8.0
 -  .NET Version: .NET Framework 4.7.2 

**Additional context**

Never had this problem before, started a couple weeks ago but I thought it was just a problem with steam so I just stopped it eventually and just forgotten all about it as IRL stuff kinda came up unexpectedly, but I'm back to wanting to get depressurizer up and going again and I want to do a scrape of the whole database instead of just owned games. Internet is not an issue nor is bandwidth an issue as neither are being hogged up nor is it going down for/at any point in time of scraping things.

[depressurizer-(14-03-2019).log](https://github.com/Depressurizer/Depressurizer/files/2969817/depressurizer-.14-03-2019.log)
That is the full log if you want to take a look at it, and see all the random errors it likes to throw.

**Comment by mvegter** [comment]
Scraping a lot of games can result in unwanted results (in general) as Steam has a limit of X page visits in X time. After this threshold has been passed it will result in either in http errors or empty store pages.

**Comment by JourneyOver** [comment]
I just find it funny how one day I can go about scanning the whole entire DB being fresh, and then the next day it starts doing the above and has been since when I'm scanning a large amount of games.

**Comment by mvegter** [comment]
I'm experiencing this too, at the moment I suspect changes to the Steam Store (scrape shield, DDoS protecting etc...)

**Comment by JourneyOver** [comment]
I feel they are def doing something, as I just tried again today and I couldn't even get it to scrap 10 games without it throwing the error...

**Comment by mvegter** [comment]
Should be fixed in the latest release!

**Comment by JourneyOver** [comment]
Nope sadly now if it hits an exception it'll throw an error and stop the scan completely xD

[depressurizer-(18-04-2019).log](https://github.com/Depressurizer/Depressurizer/files/3096158/depressurizer-.18-04-2019.log)

**Comment by mvegter** [comment]
Ugh, how many games are you attempting to scrape? And have you attempted any prior scraped in  short period of time?

**Comment by JourneyOver** [comment]
When I was attempting it yesterday, I only tried scrapping the games I owned (which equaled out to a bit over 2,100 games in total, and nope I haven't done any prior scrapping in over a few weeks.

**Comment by mike8115** [comment]
I'm running into the same issue with about 800 unscrapped games that I own. It's okay if I try a dozen or two, but it gets stuck afterwards.

**Comment by FireEqual** [comment]
The problem seems to be (at least for me) with games deleted from Steam.

For example AltCoin (AppID 782010). Other games work well, which is confirmed by scraping 40-50 games at a time. On the contrary, some games that do not exist per se in Steam, like beta servers, PUBG: Test Server (AppID 622590) works as intended. The scraping doesn't stop, it just doesn't get any info.

Also it would seem like the issue is connected with the Database too. It loads much more games than I have (1900 vs mine 1600), most of which are games I deleted from my account. Trying to delete, for example, AltCoin from the database, saving the database, unticking the option to "load games list on startup" and restarting Depressurizer with saving evaluates to nothing, for some unknown reason AltCoin is still there. I've tried reinstalling, deleting database as a whole, deleting the profile. This is what I don't get, because those games shouldn't even exist in my account file, why does Depressurizer find them?

So TL;DR the program probably gets some edge case here and loops until the timed-out exception, deleting the game from the database works only for the time the program is running; after restart, the Database loads ghost games for some unknown reason, while interestingly scraping status remains the same.

**Comment by mike8115** [comment]
I don't think AltCoin isn't deleted from Steam per se because it still has a hub and community page. Its store page gets redirected to the main store page. Depressurizer should know when this happens.

**Comment by FireEqual** [comment]
That's what I initially thought too, but for some reason I've only noticed problems with deleted games. But not all of them, Deadpool scrapes perfectly, as in doesn't get any data, but still doesn't make the operation time out. 
All of that may be just random luck for me and it may be that the program has such a problem with any game.

**Comment by mvegter** [comment]
This is part of the [Depressurizer (Premium) API](https://github.com/Depressurizer/Depressurizer#depressurizer-premium-api-), for the normal Depressurizer  using the Steam Store page scraper you will have to manually do this.
